api/src/main/java/org/pmiops/workbench/cdr/dao/CriteriaDao.java

```diff
@@ -9,7 +9,13 @@
 
 public interface CriteriaDao extends CrudRepository<Criteria, Long> {
 
-    @Query("select c from Criteria c where c.type like :type% and c.parentId = :parentId order by c.code asc")
-    List<Criteria> findCriteriaByTypeAndParentId(@Param("type") String type, @Param("parentId") Long parentId);
+    List<Criteria> findCriteriaByTypeAndParentIdOrderByCodeAsc(@Param("type") String type, @Param("parentId") Long parentId);
+
+    @Query("select c from Criteria c " +
```

Comment:
I think we're going to want to use fulltext indexes here:

https://dev.mysql.com/doc/refman/5.6/en/innodb-fulltext-index.html

In some testing I did a while back, querying with LIKE on big tables was slow, but querying with fulltext indexes was super-fast.

This is important generally, and especially important if we're going to support auto-complete as you type (which I think will be very important for a feature like this to be usable.)